pkg/servicedisc/client.go

```diff
@@ -22,31 +22,38 @@ type Config struct {
 	DiscAddr string
 }
 
-// HTTPClient is responsible for interacting with the proxy-discovery
+// HTTPClient is responsible for interacting with the service-discovery
 type HTTPClient struct {
 	log     logrus.FieldLogger
 	conf    Config
-	entry   Proxy
+	entry   Service
 	entryMx sync.Mutex // only used if UpdateLoop && UpdateStats functions are used.
 	auth    *httpauth.Client
 	client  http.Client
 }
 
 // NewClient creates a new HTTPClient.
-func NewClient(log logrus.FieldLogger, conf Config) *HTTPClient {
+func NewClient(log logrus.FieldLogger, conf Config, sType string) *HTTPClient {
```

Comment:
Instead of creating a new input, can we create a new field within the `servicedisc.Config` structure? Thanks!